### PR TITLE
snakemake: init at 5.2.2

### PR DESCRIPTION
**Pull request by SebMdB**
###### Motivation for this change

Snakemake is a workflow management system. Useful in bio-informatics and HPC, where reproducibility is appreciated.
Also add dependency: ratelimiter (init at 1.2.0.post0)

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions (Ubuntu 18.04)
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by jtojnar — COMMENTED** [review]
Please move the packages to `pkgs/development/python-modules/<name>/default.nix`, as per the comment on top of `pkgs/top-level/python-packages.nix`. It would be also nice if you could have one commit per package, and use `python.pkgs.<name>` in the commit message, instead of just `<name>`.

**Comment by Helkafen** [comment]
Hi Jan. Thanks for the review! The packages are moved to separate default.nix files and I updated the commit messages.

**Comment by ryantm** [comment]
Related: https://github.com/NixOS/nixpkgs/pull/45828

File: pkgs/development/python-modules/ratelimiter/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  doCheck = true;
+
+  buildInputs = [ pytest ];
+
+  checkPhase = ''
+    py.test tests
+  '';
+
+  meta = with stdenv.lib; {
+    homepage = "https://github.com/RazerM/ratelimiter";
```
Comment (jtojnar) [inline pkgs/development/python-modules/ratelimiter/default.nix]: Plain URLs do not need to be quoted.

File: pkgs/development/python-modules/ratelimiter/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  };
+
+  prePatch = ''
+    sed -i -e 's/\xe2//g' README.rst
+    sed -i -e 's/\x80//g' README.rst
+    sed -i -e 's/\x9c//g' README.rst
+    sed -i -e 's/\x9d//g' README.rst
+  '';
+
+  doCheck = true;
```
Comment (jtojnar) [inline pkgs/development/python-modules/ratelimiter/default.nix]: No need for explicit `doCheck` – python pakages are checked automatically.

File: pkgs/development/python-modules/ratelimiter/default.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
+{
+  stdenv
+, buildPythonPackage
+, fetchPypi
+, pytest
+}:
+
+buildPythonPackage rec {
+  name = "${pname}-${version}";
```
Comment (jtojnar) [inline pkgs/development/python-modules/ratelimiter/default.nix]: `name` is not necessary in python packages.

File: pkgs/development/python-modules/ratelimiter/default.nix [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  pname = "ratelimiter";
+  version = "1.2.0.post0";
+
+  propagatedBuildInputs = [];
```
Comment (jtojnar) [inline pkgs/development/python-modules/ratelimiter/default.nix]: Why keep this?

File: pkgs/development/python-modules/ratelimiter/default.nix [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  prePatch = ''
+    sed -i -e 's/\xe2//g' README.rst
+    sed -i -e 's/\x80//g' README.rst
+    sed -i -e 's/\x9c//g' README.rst
+    sed -i -e 's/\x9d//g' README.rst
+  '';
+
+  doCheck = true;
+
+  buildInputs = [ pytest ];
```
Comment (jtojnar) [inline pkgs/development/python-modules/ratelimiter/default.nix]: You can use `checkInputs`.

File: pkgs/development/python-modules/snakemake/default.nix [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "adffe7e24b4a613a9e8bf0a2a320b3cea236d86afb9132bb0bbbc08b8e35a3a3";
+  };
+
+  doCheck = false; # Tests depend on Google Cloud credentials at ${HOME}/gcloud-service-key.json
+
+  meta = with stdenv.lib; {
+    homepage = "http://snakemake.bitbucket.io";
+    license = "MIT";
```
Comment (jtojnar) [inline pkgs/development/python-modules/snakemake/default.nix]: Use `licenses.mit`.

File: pkgs/development/python-modules/snakemake/default.nix [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    description = "A workflow management system that aims to reduce the complexity of creating workflows by providing a fast and comfortable execution environment, together with a clean and modern specification language in python style. Snakemake workflows are essentially Python scripts extended by declarative code to define rules. Rules describe how to create output files from input files";
```
Comment (jtojnar) [inline pkgs/development/python-modules/snakemake/default.nix]: This is too long for description, move it to `longDescription` and use something shorter for `description`. You can get inspiration on [Repology](https://repology.org/metapackage/snakemake/packages).

File: pkgs/development/python-modules/ratelimiter/default.nix [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+  version = "1.2.0.post0";
+
+  propagatedBuildInputs = [];
+
+  src = fetchPypi {
+      inherit pname version;
+      sha256 = "5c395dcabdbbde2e5178ef3f89b568a3066454a6ddc223b76473dac22f89b4f7";
+  };
+
+  prePatch = ''
```
Comment (jtojnar) [inline pkgs/development/python-modules/ratelimiter/default.nix]: Unfortunately, Python < 3.7 requires `glibcLocales` in `nativeBuildInputs` and something like `LC_ALL = "en_US.UTF-8";` in the expression. But it is still better than `sed`ing.

**Review by jtojnar — CHANGES_REQUESTED** [review]

**Comment by Helkafen** [comment]
All your suggestions have been applied. The locales thing makes more sense indeed!

**Comment by Helkafen** [comment]
Added "disabledIf (!isPy3k)" because Snakemake is no longer supported on Python 2.

**Comment by renatoGarcia** [comment]
The Snakemake is more a standalone application that a Python library. IMO it should be better placed at `pkgs/applications/science/misc/snakemake/` for example that at `pkgs/development/python-modules/snakemake/`.

**Comment by ryantm** [comment]
@GrahamcOfBorg build snakemake

**Comment by ryantm** [comment]
Looks good to me; merged. I hope @renatoGarcia submits a PR to add himself as a maintainer too.

**Comment by renatoGarcia** [comment]
Thanks both of you, @ryantm and @Helkafen, for the consideration.
Helkafen, you didn't add a maintainers field on snakemake expression. Would you mind if I submit a PR with both our names like at [here](https://github.com/renatoGarcia/nixpkgs/blob/3acf6d69aa9fa4f83395a10bf4049013a3ffda47/pkgs/applications/science/misc/snakemake/default.nix#L40)?

**Comment by Helkafen** [comment]
@renatoGarcia Sure!

**Pull request merged**